### ISSUE TITLE
Determining what to do with sendClusterStateRequest, sendClusterSettingsRequest, sendLocalNodeRequest and sendRegisterRestActionsRequest

### ISSUE THREAD

**Issue by ryanbogan** [opensearch-project/opensearch-sdk-java#131]
We need to figure out how the functionality of sendClusterStateRequest, sendClusterSettingsRequest, sendLocalNodeRequest and sendRegisterRestActionsRequest are going to work for the SDK.

**Comment by owaiskazi19** [comment]
@ryanbogan the comment mentioned by @dbwiddis was related to the `static` keyword and not `startActionListener`. I have mentioned [here](https://github.com/opensearch-project/opensearch-sdk-java/pull/87#discussion_r967284241) about the other APIs as well. To frame in a better way:
Once we integrate SDK with AD Extension, we will call createIndex from SDKClient and if the request fails an exception will thrown on AD Extension end.
To have the similar exception thrown as well on OpenSearch logs we will call the `startActionListener()` API while creating Index in the catch block. I hope this clear the issue and we can close it out.

**Comment by dbwiddis** [comment]
This issue is primarily a "don't forget we need to eventually do something with this listener" signpost. Not sure anything can be done *yet* but let's not forget about it.

**Comment by owaiskazi19** [comment]
We should change this issue to address all the methods then rather just `startActionListener`. 
`sendClusterStateRequest`, `sendClusterSettingsRequest`, `sendLocalNodeRequest` and `sendRegisterRestActionsRequest`. @ryanbogan will you edit the issue then?

**Comment by dbwiddis** [comment]
I think the correct solution to this problem is to sequence all these requests inside the `ExtensionsInitRequestHandler`.   See for example how `sendRegisterRestActionsRequest()` (already done) and `sendEnvironmentSettingsRequest()` (part of #244) fits nicely into an initialization sequence:
```java
try {
    return new InitializeExtensionsResponse(extensionsRunner.settings.get(NODE_NAME_SETTING));
} finally {
    // After sending successful response to initialization, send the REST API and Settings
    extensionsRunner.setOpensearchNode(extensionsRunner.opensearchNode);
    extensionsRunner.setExtensionNode(extensionInitRequest.getExtension());
    TransportService extensionTransportService = extensionsRunner.getExtensionTransportService();
    extensionTransportService.connectToNode(extensionsRunner.opensearchNode);
    extensionsRunner.sendRegisterRestActionsRequest(extensionTransportService);
    extensionsRunner.sendRegisterCustomSettingsRequest(extensionTransportService);
    extensionsRunner.transportActions.sendRegisterTransportActionsRequest(
        extensionTransportService,
        extensionsRunner.opensearchNode,
        extensionsRunner.getUniqueId()
    );
    // Get OpenSearch Settings and set values on ExtensionsRunner
    Settings settings = extensionsRunner.sendEnvironmentSettingsRequest(extensionTransportService);
    extensionsRunner.setEnvironmentSettings(settings);
    extensionsRunner.setNamedXContentRegistry(
        new ExtensionNamedXContentRegistry(settings, extensionsRunner.getCustomNamedXContent())
    );

    // Last step of initialization
    extensionsRunner.setInitialized();
}
```
Adding the `setInitialized()` at the end as I did in #244 also resolves my concern in #94.

Recommend this issue and #94 be closed when #244 is merged.

**Comment by owaiskazi19** [comment]
@dbwiddis `sendClusterStateRequest`, `sendClusterSettingsRequest` and`sendLocalNodeRequest` are APIs which should be triggered by any extension only when it's needed. We should not add these methods in `InitializeExtensionsResponse`.
